job/strategies/templates/strategy.py

```diff
@@ -1,31 +1,39 @@
+from __future__ import annotations
+
 from abc import ABCMeta, abstractmethod
-from typing import Any, Dict, List
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, List
 
 import numpy as np
 import pandas as pd
 
 from db.mappings.model import ModelType
 from db.mappings.recommendation import Rec
-from sites.templates.site import Site
+
+if TYPE_CHECKING:
+    from sites.templates.site import Site
 
 
+@dataclass
 class Strategy(metaclass=ABCMeta):
     @abstractmethod
-    def fetch_data(site: Site, interactions_data: pd.DataFrame) -> List[Dict[str, Any]]:
+    def fetch_data(self, site: Site, interactions_data: pd.DataFrame) -> List[Dict[str, Any]]:
```

Comment:
🔧 Let's add a docstring to each of these abstract methods. We don't need to do so for each of the child classes, though!